### PR TITLE
feat(argo-cd): Upgrade Argo CD to 2.5.0

### PR DESCRIPTION
**Pull request by pdrastil**
Resolves:
- https://github.com/argoproj/argo-helm/issues/1567
- https://github.com/argoproj/argo-helm/issues/1537

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/_helpers.tpl
Comment (pdrastil) [inline charts/argo-cd/templates/_helpers.tpl:46]: Support for new HTTP / HTTPS for dex server. Tested via OIDC provider config on RC3

File: charts/argo-cd/templates/argocd-repo-server/deployment.yaml
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-repo-server/deployment.yaml:208]: Removed all these guards. Metrics ports are not related to configured metrics Service and can be used via other means like Prometheus service discovery. They are also used by some controllers for liveness / readiness probes.

File: charts/argo-cd/templates/argocd-repo-server/deployment.yaml
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-repo-server/deployment.yaml:190]: Not an optional volume. All optional volumes are marked as such via `optional: true`

**Comment by pdrastil** [comment]
Synced and tested on RC3. Still waiting for confirmation if new cmd-params for applicationset will be cherry picked to 2.5.0

File: charts/argo-cd/Chart.yaml
Comment (pdrastil) [inline charts/argo-cd/Chart.yaml:2]: To be changed to 2.5.0 when released

**Comment by karlschriek** [comment]
Hey, thanks a lot for getting this up so early. We are pretty anxious to start using 2.5.0 because of the ability to roll out Applications to namespaces other than `argocd`. I am giving this chart a spin today to see if I can get it to work correctly. 

I did notice however that the CRD rollouts in the chart are still `Namespaced`. Shouldn't they have a `Cluster` scope for the above feature to work correctly?

**Comment by pdrastil** [comment]
@karlschriek Hi - you are welcome. CRDs are always global resource in the cluster. As far as I know inside CRD schema can be namespaced (to allow creation of resources inside each namespace) or global (which means that they behave similar like ClusterRoles). This chart just installs the CRDs as part of it installation, namespaced schema ensures that it can be used anywhere.

**Comment by karlschriek** [comment]
Right! Sorry yeah my bad, Cluster CRDs are meant for CRs that don't get
rolled out to a namespace at all but are rolled out as a cluster-wide
resource, has nothing to do with where those types of CRs can be deploy.

Anyway, have rolled out and it looks like it is working. Wil post back here
if I come across any issues related to the chart specifically.

File: charts/argo-cd/Chart.yaml
Comment (jmeridth) [reply charts/argo-cd/Chart.yaml:2]: it has been done :smile: https://github.com/argoproj/argo-cd/releases/tag/v2.5.0

Comment (pdrastil) [reply charts/argo-cd/Chart.yaml]: @jmeridth Builiding

**Review by jmeridth — APPROVED** [review]
This LGTM but would like another approval before merge

**Review by hairmare — APPROVED** [review]
I tried this on a recent kind v1.24.0 and didn't run into any issues using the out-of-the-box values.

This is just just a drive-by LGTM since i'm not a maintainer though.

**Review by mkilchhofer — COMMENTED** [review]
Additional to the comment, we should cleanup the default securityContexts as upstream already has default (hardened) security context defined.
And release 2.5 now adds 
```yaml
          seccompProfile:
            type: RuntimeDefault
```
Also the TLS stuff for dex is missing. But I assume you want to do this outside the 2.5 PR and test it again via:
- https://github.com/argoproj/argo-helm/pull/1477

But I did not find a blocker. So I am okay releasing this one and fix the

**Comment by pdrastil** [comment]
@mkilchhofer - See following

Ad security contexts - already have local branch with this because it needs more work on all components.
Dex TLS - The protocol now uses HTTPS by default, certs are self-signed. Custom certs will follow here https://github.com/argoproj/argo-helm/pull/1477
Probes - I'll check in next iteration - they were still present in RC3

**Comment by bobertrublik** [comment]
Thank you for your work! The argocd-server serviceaccount still needs more permissions when apps are deployed to a different namespace (as per https://github.com/argoproj/argo-cd/pull/9755#issuecomment-1212915993).
I'm however not sure what the best practices are here. Can we grant the permissions to the ClusterRole?

**Comment by pdrastil** [comment]
I believe it should be ClusterRole as you are using different namespace. I will take a look what needs to be patched and add it as another fix.

**Pull request merged**
